bin/tx-prover/src/server/mod.rs

```diff
@@ -0,0 +1,87 @@
+use generated::api::{api_server, ProveTransactionRequest, ProveTransactionResponse};
+use miden_objects::transaction::{ProvenTransaction, TransactionWitness};
+use miden_prover::DeserializationError;
+use miden_tx::{
+    utils::{Deserializable, Serializable},
+    LocalTransactionProver, TransactionProver,
+};
+use tokio::{net::TcpListener, sync::Mutex};
+use tonic::{Request, Response, Status};
+use tracing::info;
+
+pub mod generated;
+
+pub struct Rpc {
+    pub api_service: api_server::ApiServer<RpcApi>,
+    pub listener: TcpListener,
+}
+
+impl Rpc {
+    pub fn new(listener: TcpListener) -> Self {
+        let api_service = api_server::ApiServer::new(RpcApi::default());
+        Self { listener, api_service }
+    }
+}
+
+#[derive(Default)]
+pub struct RpcApi {
+    prover: Mutex<LocalTransactionProver>,
+}
+
+// We need to implement Send and Sync for the generated code to be able to use the prover in the
+// shared context.
+unsafe impl Send for RpcApi {}
+unsafe impl Sync for RpcApi {}
```

Comment:
I'll merge as is, but ideally, the comment should explain why it is OK to implement `Send` and `Sync`.